src/ui/map.js

```diff
@@ -2167,7 +2184,50 @@ class Map extends Camera {
      * @returns {Object} terrain Terrain specification properties of the style.
      */
     getTerrain(): Terrain | null {
-        return this.style.getTerrain();
+        return this.style ? this.style.getTerrain() : null;
+    }
+
+    /**
+     * Sets the fog property of the style.
+     * @param fog The fog properties to set. Must conform the [Mapbox Style Specification](https://docs.mapbox.com/mapbox-gl-js/style-spec/root/#fog).
+     * If `null` or `undefined` is provided, this function call removes the fog from the map.
+     * @returns {Map} `this`
+     * @example
+     * map.setFog({
+     *  "range": [1.0, 12.0],
+     *  "color": 'white',
+     *  "horizon-blend": 0.1
+     * });
+     */
+    setFog(fog: FogSpecification) {
+        this._lazyInitEmptyStyle();
+        this.style.setFog(fog);
+        return this._update(true);
+    }
+
+    /**
+     * Returns the fog specification or `null` if fog is not set on the map.
+     *
+     * @returns {Object} fog Fog specification properties of the style.
+     */
+    getFog(): Fog | null {
+        return this.style ? this.style.getFog() : null;
+    }
+
+    /**
+     * Returns the fog opacity for a given geolocation.
+     *
+     * An opacity of 0 means that there is no fog contribution for the given geolocation
+     * while a fog opacity of 1.0 means the geolocation is fully obscured by the fog effect.
+     *
+     * If there is no fog set on the map, this function will simply return 0.
```

Comment:
nit: ~~simply~~

Reply:
Addressed in https://github.com/mapbox/mapbox-gl-js/pull/10564/commits/e5abf7594f814468e622f6953806216c1c179389